src/lib/Persistor.js

```diff
@@ -2,35 +2,50 @@
 
 var _ = require('underscore'),
     Q = require('q'),
-    Class = require('class.extend');
+    Class = require('class.extend'),
+    low = require('lowdb'),
+    FileSync = require('lowdb/adapters/FileSync');
 
 module.exports = Class.extend({
 
    init: function() {
-      // TODO: actually read/write to a file
-      this.inMemoryStorage = {};
+      this.adapter = new FileSync('./store/db.json');
+      this.db = low(this.adapter);
+      this.db.defaults({})
+         .write();
    },
 
    put: function(hash, range, value) {
-      if (!_.isObject(this.inMemoryStorage[hash])) {
-         this.inMemoryStorage[hash] = {};
+      var currentHash = this.db.get(hash).value(),
+          newObj = {};
+
+      if (!_.isObject(currentHash)) {
+         this.db.set(hash, {})
+            .write();
       }
 
-      this.inMemoryStorage[hash][range] = value;
+      newObj[range] = value;
+
+      this.db.get(hash)
+         .assign(newObj)
+         .write();
 
       return Q.when();
    },
 
    get: function(hash, range) {
-      if (_.isUndefined(range)) {
-         return Q.when(this.inMemoryStorage[hash]);
+      var currentHash = this.db.get(hash).value(),
+          withRange = currentHash[range];
+
+      if (!_.isUndefined(withRange)) {
```

Comment:
Hmmmm, doesn't this have the same issue as noted [here](https://github.com/onebytegone/leaderboard-server/pull/2#discussion_r215621470)?

Reply:
If "range" *is* defined on the stored object, then it will return that, else...

...it will check for current hash, and return that, else...

...it will return undefined.

With this new implementation, there is no situation where the wrong data will be returned when something is undefined.

However in testing this through I see now that the script will choke when currentHash is undefined cuz I done forgot to check it before trying to assign currentHash[range]....

Reply:
Ah, you're right. 🤦‍♂️ Was having problems parsing "Wait, but `range` is optional... Why is it being used?".